eloquentjs/5_4_dominant_writing_direction.js

```diff
@@ -0,0 +1,118 @@
+const SCRIPTS = require('./bin/scripts.js');
+
+/**
+ * Returns a string indicating the dominant writing direction present in an input string. 
+ * If there is a tie between dominant writing directions its resolved by alphabetic ordering of direction strings(ltr < rtl < ttb).
+ * @param {*} inputString 
+ * @returns 
+ */
+function dominantWritingDirection(input) {
+    input = (input.split('').filter(b => scriptFromCharacter(b) !== null)).join(''); // filters out invalid characters
+
+    let directionGroups = countBy(input, j => scriptFromCharacter(j).direction);
+
+    directionGroups.sort((a, b) => {
+        if (a.count < b.count) {
+            return 1;
+        }
+        if (a.count > b.count) {
+            return -1;
+        }
+        if (a.count === b.count) {
+            return a.name[0] > b.name[0] ? 1 : -1;
+        }
+        return 0;
+    });
+
+    return directionGroups[0].name;
+}
+
+/**
+ * Returns the script that corresponds to an input character.
+ * (Modifed helper function pulled from eloquent JS chapter 5)
+ * @param {*} input 
+ * @returns 
+ */
+function scriptFromCharacter(input) {
+    const code = input.codePointAt(0);
+
+    for (let script of SCRIPTS) {
+        if (script.ranges.some(([from, to]) => {
```

Comment:
I'd pull the logic in this if statement out into it's own const to make the control flow clearer